packages/expo-av/ios/EXAV/EXAV.m

```diff
@@ -595,14 +601,18 @@ - (int)_getDurationMillisOfRecordingAudioRecorder
 - (NSDictionary *)_getAudioRecorderStatus
 {
   if (_audioRecorder) {
-    int durationMillisFromRecorder = [self _getDurationMillisOfRecordingAudioRecorder];
-    // After stop, the recorder's duration goes to zero, so we replace it with the correct duration in this case.
-    int durationMillis = durationMillisFromRecorder == 0 ? _audioRecorderDurationMillis : durationMillisFromRecorder;
+    // [_audioRecorder currentTime] returns bad values after changing to bluetooth input
+    // so we track durationMillis independently of the audio recorder.
+    // see: https://stackoverflow.com/questions/43351904/avaudiorecorder-currenttime-giving-bad-values
+    int curTimestamp = (int) (_audioRecorder.deviceCurrentTime * 1000);
+    int curDuration = [_audioRecorder isRecording] ? (curTimestamp - _audioRecorderStartTimestamp) : 0;
+    int durationMillis = _prevAudioRecorderDurationMillis + curDuration;
```

Comment:
I unfortunately had to keep track of recording duration manually after making this change. If recording inputs are changed while a recording is in progress, I found that the reported `currentTime` of the recording became buggy, and reported ridiculously large numbers.